test/daemon_test.rb

```diff
@@ -0,0 +1,104 @@
+require 'test_helper'
+require 'active_support'
+require 'mocha/mini_test'
+require 'logging'
+require 'dynflow/testing'
+require_relative '../lib/dynflow/rails'
+
+class DaemonTest < ActiveSupport::TestCase
+  setup do
+    @dynflow_memory_watcher = mock('memory_watcher')
+    @daemons = mock('daemons')
+    @daemon = ::Dynflow::Rails::Daemon.new(
+      @dynflow_memory_watcher,
+      @daemons
+    )
+    @world_class = mock('dummy world factory')
+    @dummy_world = ::Dynflow::Testing::DummyWorld.new
+    @dummy_world.stubs(:auto_execute)
+    @event = Concurrent.event
+    @dummy_world.stubs(:terminated).returns(@event)
+    @world_class.stubs(:new).returns(@dummy_world)
+    @dynflow = ::Dynflow::Rails.new(
+      @world_class,
+      ::Dynflow::Rails::Configuration.new
+    )
+    ::Rails.stubs(:application).returns(OpenStruct.new(:dynflow => @dynflow))
+    ::Rails.stubs(:root).returns('support/rails')
+    ::Rails.stubs(:logger).returns(Logging.logger(STDOUT))
+    @dynflow.require!
+    @daemon.stubs(:sleep).returns(true) # don't pause the execution
+    @current_folder = File.expand_path('../support/rails/', __FILE__)
+    ::ActiveRecord::Base.configurations = { 'development' => {} }
+    ::Dynflow::Rails::Configuration.any_instance.stubs(:initialize_persistence).
+      returns(WorldFactory.persistence_adapter)
+  end
+
+  teardown do
+    @event.complete
```

Comment:
Had to make sure there were no zombies by the end of testing in order to return 0